coriolis_openstack_utils/actions/network_actions.py

```diff
@@ -390,3 +401,133 @@ def execute_operations(self):
     def cleanup(self):
         routers.delete_router(
             self._destination_openstack_client, self.get_new_router_name())
+
+
+class PortCreationAction(base.BaseAction):
+    """ Action class for replicating Neutron ports.
+    param action_payload: dict(): payload (params) for the action
+    must contain 'src_port_id'
+                 'dest_network_id'
+    """
+
+    action_type = base.ACTION_TYPE_CHECK_CREATE_ROUTER
+    NEW_ROUTER_DESCRIPTION_FORMAT = (
+        "Created by the Coriolis OpenStack utilities for source router '%s'.")
+
+    @property
+    def relevant_keys(self):
+        relevant_keys = [
+            'allowed_address_pairs', 'extra_dhcp_opts',
+            'binding:profile', 'admin_state_up',
```

Comment:
The official Neutron API docs say you can https://developer.openstack.org/api-ref/network/v2/?expanded=create-port-detail#ports defining `binding: profile` as a "dictionary that enables the application running on the specific host to pass and receive vif port information specific to the networking back-end" with no specific format being defined, `binding:vnic_type` can have the following values: `[normal, macvtap, direct, baremetal, direct-physical, virtio-forwarder]`. Both values are optional values in the request body for creating a port according to the docs.

Reply:
Thanks for the reply, `binding:profile` can stay considering it's both optional and opaque but please skip `binding:vnic_type` in case the source/dest env provider network setups don't "line up" in order for it to be preserved.